photutils/psf/photometry.py

```diff
@@ -0,0 +1,427 @@
+                             'into an integer.')
+    
+    @property
+    def fitshape(self):
+        return self._fitshape
+
+    @fitshape.setter
+    def fitshape(self, value):
+        value = np.asarray(value)
+        if value.size == 2:
+            if np.all(value) > 0:
+                if np.all(value % 2) == 1:
+                    self._fitshape = tuple(value)
+                else:
+                    raise ValueError('fitshape must be odd integer-valued, '
+                                     'received fitshape = {}'\
+                                     .format(value))
+            else:
+                raise ValueError('fitshape must have positive elements, '
+                                 'received fitshape = {}'\
+                                 .format(value))
+        else:
+            raise ValueError('fitshape must have two dimensions, '
+                             'received fitshape = {}'.format(value))
+
+    @property
+    def aperture_radius(self):
+        return self._aperture_radius
+
+    @aperture_radius.setter
+    def aperture_radius(self, value):
+        if isinstance(value, (int, float)) and value > 0:
+            self._aperture_radius = value
+        elif value is None:
+            self._aperture_radius = value
+        else:
+            raise ValueError('aperture_radius must be a real-valued '
+                             'number, received aperture_radius = {}'
+                             .format(value))
+
+
+    def __call__(self, image, positions=None):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Image to perform photometry.
+        positions : `~astropy.table.Table` (optional)
+            Positions, in pixel coordinates, at which stars are located.
+            The columns must be named as 'x_0' and 'y_0'. 'flux_0' can also
+            be provided to set initial fluxes.
+
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+        
+        return self.do_photometry(image, positions)
+
+    def do_photometry(self, image, positions=None):
+        """
+        Perform PSF photometry in ``image``. This method assumes that
+        ``psf`` has centroids and flux parameters which will be fitted to the
+        data provided in ``image``. A compound model, in fact a sum of
+        ``psf``, will be fitted to groups of starts automatically identified
+        by ``group``. Also, ``image`` is not assumed to be background
+        subtracted.
+        If positions are not ``None`` then this method performs forced PSF
```

Comment:
I know we discussed it already, but I don't think that "forced PSF photometry" is a well-known term, e.g. I don't know what exactly that means. So there should e an extra sentence here to explain what is "forced"